django/db/models/sql/compiler.py

```diff
@@ -150,7 +147,14 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr.output_field, 'model') == self.query.model):
                     pk = expr
```

Comment:
I know this isn't part of your code, but can we break out of the loop if this condition is met?